logical/framework/field_data_test.go

```diff
@@ -467,6 +468,40 @@ func TestFieldDataGet(t *testing.T) {
 			},
 		},
 
+		"type header, keypair string array": {
+			map[string]*FieldSchema{
+				"foo": {Type: TypeHeader},
+			},
+			map[string]interface{}{
+				"foo": []interface{}{"key1=value1", "key2=value2", "key3=1"},
+			},
+			"foo",
+			http.Header{
+				"Key1": []string{"value1"},
+				"Key2": []string{"value2"},
+				"Key3": []string{"1"},
+			},
+		},
+
+		"type header, map string slice": {
+			map[string]*FieldSchema{
+				"foo": {Type: TypeHeader},
+			},
+			map[string]interface{}{
+				"foo": map[string][]string{
```

Comment:
I think the JSON Unmarshaller will return this as a `map[string][]interface{}`, can you test some non-string value and make sure you get what you expect.

Reply:
Will do.

Reply:
It was a good test! 

I ran Vault in dev mode and made a little path that took `TypeHeader` for a field named `headers`. Then I used the following command:
```